### PR TITLE
feat: implement `LENGTH` function

### LINKED ISSUE #1297: Implement `LENGTH` function
# Description

The method should be able to receive `String`, `Map`, and `List` types.

# Example

```
gluesql> SELECT LENGTH('ABC');

| LENGTH('ABC') |
|---------------|
| 3             |

gluesql> SELECT LENGTH(CAST('[1, 2, 3]' AS LIST));
| LENGTH(CAST('[1, 2, 3]' AS LIST)) |
|-----------------------------------|
| 3                                 |

gluesql> SELECT LENGTH(CAST('{"a": 1, "b": 5, "c": 9, "d": 10}' AS MAP));
| LENGTH(CAST('{"a": 1, "b": 5, "c": 9, "d": 10}' AS MAP)) |
|----------------------------------------------------------|
| 4                                                        |
```

### PR DESCRIPTION
**Pull request by jopemachine**
Fixes #1297.

### CODE REVIEW

File: test-suite/src/function/length.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use gluesql_core::prelude::Value;
+
+use {crate::*, gluesql_core::prelude::Payload};
+
+test_case!(length, async move {
+    test! {
+        name: "test length with string",
+        sql: "SELECT LENGTH('Hello.');",
+        expected: Ok(Payload::Select { labels: vec!["LENGTH('Hello.')".to_owned()], rows: vec![vec![Value::U64(6)]] })
```
Comment (ever0de) [inline test-suite/src/function/length.rs:9]: Let's use the `select!` macro to write the expected part like any other test case.

Comment (ever0de) [reply test-suite/src/function/length.rs:9]: https://github.com/gluesql/gluesql/blob/29f4128d38a11848cf123c60dd6b9b2f4076cbd7/test-suite/src/basic.rs#L44-L51

**Review by ever0de — CHANGES_REQUESTED** [review]
https://coveralls.io/builds/61364317

Please refer to the coveralls and write test cases for the uncovered parts.

**Review by ever0de — APPROVED** [review]
Looks good to me, what does anyone else think?

**Review by panarch — APPROVED** [review]
Looks all great! Thanks a lot for the contribution 👍 👍

**Pull request merged**